Add support for Zhipu AI GLM-4.5 LLM

CaptainJi opened the pull request:
feat: Add support for Zhipu AI GLM-4.5 model
Integrated support for the Zhipu AI GLM-4.5 large language model.

fix: Changes Based on Feedback
Reset .gitignore file.
Renamed the class name from GLM_Chat to GLM_4_5 to align with the specific model version.
Updated the verify_environment() return value from "GEMINI_API_KEY" to "GLM_API_KEY" for naming consistency.
Signed-off-by: Ji Qing jiqing19861123@163.com

kichanyurd commented on src/parlant/adapters/nlp/glm_service.py:
```diff
@@ -0,0 +1,224 @@
+
+        if not os.environ.get("GLM_API_KEY"):
+            return """\
+You're using the GLM NLP service, but GEMINI_API_KEY is not set.
```
Need to change "GEMINI_API_KEY" to "GLM_API_KEY"

kichanyurd commented:
Hi @CaptainJi and thank you for this contribution.

I left a few review comments.

Also, before we can merge this, we need to ensure this conforms to our contribution policy and DCO.

Your current commits are signed-off with a nickname ("CaptainJi"), but our contribution policy requires you to set your first and last name, as well as email, in your git, so as to sign-off the content in an authorized manner. Please review this file for details: https://github.com/emcie-co/parlant/blob/develop/CONTRIBUTING.md

So just a few more small steps and we'll be ready to merge your contribution, mate.

CaptainJi commented:
Hi @kichanyurd , thank you for the detailed review!

I've addressed all your feedback:
- Restored the missing `.gitignore` file
- Renamed the main class from `GLM_Chat` to `GLM_4_5`
- Updated the API key constant in `verify_environment()` return value
- Corrected author information and added DCO sign-off

The PR has been updated with these changes. Please review when convenient and let me know if anything requires further refinement. 👍

CaptainJi commented:
> Hi @kichanyurd , thank you for the detailed review!
> 
> I've addressed all your feedback:
> 
> * Reset `.gitignore` file
> * Renamed the main class from `GLM_Chat` to `GLM_4_5`
> * Updated the API key constant in `verify_environment()` return value
> * Corrected author information and added DCO sign-off
> 
> The PR has been updated with these changes. Please review when convenient and let me know if anything requires further refinement. 👍

kichanyurd commented:
@CaptainJi thank you - merged!
